src/slider/slider.js

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import * as React from 'react';
+import {findDOMNode} from 'react-dom';
+import {thumbWidth} from './constants';
+import type {StatelessStateT, PropsT} from './types';
+import {
+  Root as StyledRoot,
+  Axis as StyledAxis,
+  AxisRange as StyledAxisRange,
+  Tick as StyledTick,
+  TickBar as StyledTickBar,
+  Thumb as StyledThumb,
+} from './styled-components';
+import {getOverrides} from '../helpers/overrides';
+/* global document */
+/* global window */
+
+class Slider extends React.Component<PropsT, StatelessStateT> {
+  domNode: ?Element | Text;
+
+  static defaultProps = {
+    overrides: {},
+    onChange: () => {},
+    onAxisClick: () => {},
+    onThumbDown: () => {},
+    onMouseMove: () => {},
+    onThumbUp: () => {},
+    error: false,
+    autoFocus: false,
+    tabIndex: 0,
+  };
+
+  state = {
+    isThumbMoving: false,
+    currentThumb: -1,
+    currentMove: 0,
+    thumbRefs: this.props.value.map(() => React.createRef()),
+  };
+
+  componentDidMount() {
+    /* eslint-disable-next-line react/no-find-dom-node */
+    this.domNode = findDOMNode(this);
+  }
+
+  onThumbDown = (event: MouseEvent, thumbIndex: number) => {
+    document.addEventListener('mousemove', this.onMouseMove);
+    document.addEventListener('mouseup', this.onThumbUp);
+    this.setState({
+      isThumbMoving: true,
+      currentThumb: thumbIndex,
+    });
+    this.props.onThumbDown({
+      event,
+      currentThumb: thumbIndex,
+      isThumbMoving: true,
+    });
+  };
+
+  onThumbUp = (event: MouseEvent) => {
+    const {isThumbMoving} = this.state;
+    if (isThumbMoving) {
+      this.setState({
+        isThumbMoving: false,
+        currentThumb: -1,
+        currentMove: 0,
+      });
+    }
+    this.props.onThumbUp({
+      event,
+      isThumbMoving,
```

Comment:
If we want to pass this value in here its value should be set to `false` explicitly since the state's value won't be changed yet: `isThumbMoving: false`

Reply:
removed from API and cleaned it up